packages/api-reference/src/features/ExampleRequest/ExampleRequest.vue

```diff
@@ -120,40 +108,114 @@ const hasMultipleExamples = computed<boolean>(
 
-const getGlobalSecurity = inject(GLOBAL_SECURITY_SYMBOL)
+/** Current request */
+const request = computed(() => {
+  const requestsArr = Object.values(requests)
+  if (!requestsArr.length) return null
+
+  return requestsArr.find(
+    ({ method, path }) =>
+      method === props.operation.httpVerb.toLowerCase() &&
+      path === props.operation.path,
+  )
+})
+
+/** Just grab the first example for now */
+const requestExample = computed(() =>
+  request.value ? requestExamples[request.value.examples[0]] : null,
+)
+
+/** Selected security scheme uids restircted by what is required by the request */
+const selectedSecuritySchemeUids = computed(() => {
+  // Grab the required uids
+  const requirementObjects =
+    request.value?.security ?? activeCollection.value?.security ?? []
+  const filteredObjects = requirementObjects.filter(
+    (r) => Object.keys(r).length,
+  )
+  const filteredKeys = filteredObjects.map((r) => Object.keys(r)[0])
 
-async function generateSnippet() {
+  // We have an empty object so any auth will work
+  if (filteredObjects.length < requirementObjects.length) {
+    return activeCollection.value?.selectedSecuritySchemeUids ?? []
+  }
+  // Otherwise filter the selected uids by what is required by this request
+  else {
+    const _securitySchemes = Object.values(securitySchemes)
+    const requirementUids = filteredKeys
+      .map((name) => _securitySchemes.find((s) => s.nameKey === name)?.uid)
+      .filter(Boolean)
+
+    return activeCollection.value?.selectedSecuritySchemeUids.filter((uid) =>
+      requirementUids.find((fuid) => fuid === uid),
+    )
+  }
+})
+
+/** Generates an array of secrets we want to hide in the code block */
+const secretCredentials = computed(() =>
+  selectedSecuritySchemeUids.value?.flatMap((uid) => {
+    const scheme = securitySchemes[uid]
+    if (scheme?.type === 'apiKey') return scheme.value
+    if (scheme?.type === 'http')
+      return [
+        scheme.token,
+        scheme.password,
+        btoa(`${scheme.username}:${scheme.password}`),
+      ]
+    if (scheme?.type === 'oauth2')
+      return Object.values(scheme.flows).map((flow) => flow.token)
+
+    return []
+  }),
+)
+
+const generateSnippet = async () => {
   // Use the selected custom example
   if (localHttpClient.value.targetKey === 'customExamples') {
     return (
       customRequestExamples.value[localHttpClient.value.clientKey]?.source ?? ''
     )
   }
+  if (!request.value || !requestExample.value) return ''
 
   // Generate a request object
-  const harRequest = getHarRequest(
-    {
-      url: getUrlFromServerState(serverState).modifiedUrl,
-    },
-    getRequestFromOperation(
-      props.operation,
-      {
-        replaceVariables: true,
-      },
-      selectedExampleKey.value,
-      true,
-    ),
-    getRequestFromAuthentication(
-      authenticationState,
-      props.operation.information?.security ?? getGlobalSecurity?.(),
-    ),
+  const [error, response] = createRequestOperation({
```

Comment:
Use the same request that we send from the client now, the only thing its missing is content-type